freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/FreeIpaClientFactory.java

```diff
@@ -228,10 +229,10 @@ private FreeIpaClientBuilder getFreeIpaClientBuilderForDirectMode(Stack stack, I
         return new FreeIpaClientBuilder(ADMIN_USER, freeIpa.getAdminPassword(), httpClientConfig, gatewayPort, instanceMetaData.getDiscoveryFQDN(), tracer);
     }
 
-    private FreeIpaClientException createFreeIpaStateIsInvalidException(Status stackStatus) {
+    private InvalidFreeIpaStateException createFreeIpaStateIsInvalidException(Status stackStatus) {
         String message = String.format("Couldn't build FreeIPA client. Because FreeIPA is in invalid state: '%s'", stackStatus);
         LOGGER.warn(message);
-        return new FreeIpaClientException(message, new FreeIpaHostNotAvailableException(message));
+        return new InvalidFreeIpaStateException(message, new FreeIpaHostNotAvailableException(message));
```

Comment:
1. I am pretty sure this doesn't interfere with the retries, so it should be good. i.e.
```
@Retryable(value = RetryableFreeIpaClientException.class,
            maxAttemptsExpression = RetryableFreeIpaClientException.MAX_RETRIES_EXPRESSION,
            backoff = @Backoff(delayExpression = RetryableFreeIpaClientException.DELAY_EXPRESSION,
                    multiplierExpression = RetryableFreeIpaClientException.MULTIPLIER_EXPRESSION))
```
should work since this isn't one of the cases where FreeIpaClientExceptions can be converted/wrapped with RetryableFreeIpaClientException.


2. Did you consider line 130? I think this might also need to have to switch to the InvalidFreeIpaStateException.
```
                return client.orElseThrow(() -> createFreeIpaUnableToBuildClient(new FreeIpaHostNotAvailableException("No FreeIPA client was available")));
```

Reply:
2. for now we would like to handle those cases separately where the status of the FreeIPA is already known to be wrong. I don't think I have seen this issue yet. Still, I removed the callt ot `createFreeIpaUnableToBuildClient` as there is a catch reight after which already calls that